### ISSUE TITLE
Request for comment: re: future of TF development on macOS.

### ISSUE THREAD

**Issue by willgryan** [apple/tensorflow_macos#274]
Are there plans to merge this #250 or any info regarding the future of TensorFlow development on macOS (especially M1 support)? The repo has been inactive for nearly 3 months, which is more than half the time that M1 Macs have been available. 

While it's understood that this is a dev release, TF is largely unusable on macOS especially M1 Macs.

Re: these features currently not being supported:
- tf.vectorized_map
- Higher-order gradients
- Jacobian-vector products (aka. forwardprop)

Are there plans to support these TF features? W/O these, imo only basic use cases of TF are possible.

**Comment by amirhosseingholami** [comment]
I think apple's team should pay more attention to this repo :)
all developers who bought M1 mac wants to use advantages of it!!

**Comment by Tioz90** [comment]
I really hope they'll address this at WWDC.

**Comment by jbmaxwell** [comment]
Yeah, I had high hopes for S4TF, since it _seemed_ to represent a strong path forward for ML on macOS... until Google killed it, that is... Now I feel like the whole native ML landscape for macOS is quite unclear and confusing, with CoreML, coremltools, CreateML, etc. Even if Apple would just pick up where S4TF left off, so we could feel confident devoting our time to learning it, that would be great. As it stands right now I don't feel like investing time into anything other than PyTorch (or TF) is really a good use of my time.
I'm not on M1 yet, but will hopefully be switching soon (once a 16" MacBook Pro is available).

**Comment by Yannik1337** [comment]
Is there any way to install the mentioned #250 manually, as long as it's not merged by Apple?

**Comment by tzakharko** [comment]
I would also like to know what is the status of this. It was my impression that the ML compute implementation was supposed to be merged into the upstream, but it's now six month later and there has been no development whatsoever. Apple, your new hardware is nothing short of impressive, but you _need_ to up your game in software support to promote adoption.

**Comment by maccloud42** [comment]
I'm sitting on an investment for the last 3 months that I cant use for my original purpose. Not impressed with continued silence Apple. If you bring a new chip architecture it needs to be supported. 3 months of quiet is really bad. Wont be investing in stuff I cant use again.

**Comment by aqrln** [comment]
Apparently this is the future, I guess: https://developer.apple.com/metal/tensorflow-plugin/

**Comment by jbmaxwell** [comment]
hmmm... yeah... the WWDC keynote was a complete bust for any sign of progress on the ML front. Hopefully some of the sessions this week will shed some light.

**Comment by swittk** [comment]
> Apparently this is the future, I guess: https://developer.apple.com/metal/tensorflow-plugin/

Nice find!
